clumper/decorators.py

```diff
@@ -58,3 +60,59 @@ def wrapped(clumper, *args, **kwargs):
         return method(clumper, *args, **kwargs)
 
     return wrapped
+
+
+def multifile(param_name="path"):
+    def decorator(f):
+        sig = inspect.signature(f)
+
+        if param_name not in sig.parameters:
+            raise ValueError(f"Wrapped function has no parameter '{param_name}'")
+
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            bound_arguments = sig.bind(*args, **kwargs)
+
+            bound_arguments.apply_defaults()
+
+            path = bound_arguments.arguments[param_name]
+
+            # Using this path.Glob, extract all the files if any
+            collected_clumpers = []
+            # If path is provided as string
+            if isinstance(path, str):
+                if path.startswith("https:") or path.startswith("http:"):
+                    return f(*args, **kwargs)
+
+                if "*" not in path:
+                    return f(*args, **kwargs)
+                else:
+                    path = pathlib.Path().glob(path)
+            # If path is provided as PosixPath
+            if isinstance(path, pathlib.PosixPath):
+                return f(*args, **kwargs)
+
+            # Else path must be a iterable i.e glob or array
+            for p in path:
+                # Set the path variable to p
+                bound_arguments.arguments[param_name] = str(p)
+                # Call the reader function and create the Clumper object
+                clumper = f(*bound_arguments.args, **deepcopy(bound_arguments.kwargs))
+                collected_clumpers.append(clumper)
+
+            if len(collected_clumpers) == 1:  # Only one object found
+                return collected_clumpers[0]
+
+            elif len(collected_clumpers) > 1:  # More than one object found
+                # Combine them by concating their dict
+                blob = reduce(
+                    lambda a, b: a + b, [c.collect() for c in collected_clumpers]
```

Comment:
`.concat` might be nicer here

Reply:
This is a really good tip! I didn't know that this function existed.